### PR TITLE
Sync request parser

### PR DESCRIPTION
**Pull request by phillipm**
Add parser for synchronous request. 

Synchronous requests will be used to claim cases from the server in a synchronous transaction.

Example suite xml entry:
```
<sync-request>
    <post url="fake.com/claim_patient/">
      <data key="selected_name" ref="instance('patients')/case[@case_type='test_case'][@status='open'][0]/name"/>
      <data key="selected_case_id" ref="instance('patients')/case[@case_type='test_case'][@status='open'][0]/@case_id"/>
    </post>
    <instance id="casedb" src="jr://instance/casedb"/>
    <command id="patient-search">
      <display>
        <text>Global search for person</text>
      </display>
    </command>
    <session>
      <query url="fake.com/patient_search/" storage-instance="patients">
        <data key="device_id" ref="instance('session')/session/data/uuid"/>
        <data key="device_case_count" ref="instance('session')/session/data/case_count"/>
        <prompt key="name">
          <display>
            <text>Input patient name</text>
          </display>
        </prompt>
        <prompt key="patient_id">
          <display>
            <text>Input patient id</text>
          </display>
        </prompt>
      </query>
      <datum id="case_id" nodeset="instance('patients')/case[@case_type='geriatric'][@status='open']" value="./@case_id" detail-select="m0_case_short" detail-confirm="m0_case_long"/>
    </session>
    <stack>
      <push>
        <command value="'m1-f0'"/>
        <datum id="calculated_data" value="'claimed'"/>
      </push>
    </stack>
  </sync-request>
```

### CODE REVIEW

File: tests/resources/app_structure/suite.xml [hunk 1]
```diff
@@ -61,6 +61,43 @@
       <datum id="case_id" nodeset="instance('casedb')/casedb/case[@case_type='test_case'][@status='open']" value="./@case_id" detail-select="m0_case_short" detail-confirm="m0_case_long"/>
     </session>
   </entry>
+
+  <sync-request>
```
Comment (phillipm) [inline tests/resources/app_structure/suite.xml]: Basic test that the parser works. More tests will come when the objects parsed start actually getting used.

File: backend/src/org/commcare/xml/EntryParser.java [hunk 2]
```diff
@@ -97,10 +94,69 @@ public Entry parse() throws InvalidStructureException, IOException, XmlPullParse
...
+        if (FORM_ENTRY_TAG.equals(parserBlockTag)) {
             return new FormEntry(commandId, display, data, xFormNamespace, instances, stackOps, assertions);
-        } else {
+        } else if (VIEW_ENTRY_TAG.equals(parserBlockTag)) {
             return new ViewEntry(commandId, display, data, instances, stackOps, assertions);
+        } else if (SYNC_REQUEST_TAG.equals(parserBlockTag)) {
+            return new SyncEntry(commandId, display, data, instances, stackOps, assertions, post, queries);
+        }
+
+        throw new RuntimeException("Misconfigured entry parser");
```
Comment (orangejenny) [inline backend/src/org/commcare/xml/EntryParser.java]: Nitpick, but this could be more useful by including what the given parserBlockTag was.

**Comment by orangejenny** [comment]
Looks great.

**Comment by orangejenny** [comment]
Can you update https://github.com/dimagi/commcare/wiki/Suite20 when you have a chance?

**Comment by proteusvacuum** [comment]
:mag: :information_desk_person:

**Pull request merged**